### PR TITLE
Minor changes to problem makefiles and OMP fix

### PR DESCRIPTION
**Pull request by stefanhenneking**
- Updated various problem `makefile`s:
   - include preprocessor variables
   - remove deprecated parts
- Minor OMP fix in problem implementations:
   - `set_environment`, `main`
   - only set/get `omp_num_threads` if `HP3D_USE_OPENMP` is set

### CODE REVIEW

**Comment by stefanhenneking** [comment]
@jbadger95 can you take a quick look. I updated the makefiles and fixed some omp issues for users who don't have openmp support which could cause problems when not excluding the omp calls with the preprocessor variable.

File: trunk/problems/MAXWELL/GALERKIN/makefile [hunk 1]
```diff
@@ -4,27 +4,33 @@
...
+HP3D_USE_OPENMP    ?= 0
+HP3D_USE_INTEL_MKL ?= 0
 
 ifeq ($(COMPLEX),0)
 $(error COMPLEX FLAG MUST BE 1 FOR MAXWELL APPLICATION)
 endif
 
+ifeq ($(COMPILER),INTEL)
+ifeq ($(DEBUG),1)
+FFLAGS += -gen-interface -warn none -warn interfaces
```
Comment (jbadger95) [inline trunk/problems/MAXWELL/GALERKIN/makefile:16]: I also add `-check bounds` to the debugging flags

**Review by jbadger95 — APPROVED** [review]
Looks good to me, I'm assuming you tested them, right?

File: trunk/problems/LASER/outputs/vtk-fields.dat [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Signal:
```
[the remark this answers is not in the document]
Comment (stefanhenneking) [reply trunk/problems/LASER/outputs/vtk-fields.dat:1]: definitely not :-) thanks for flagging it

**Comment by stefanhenneking** [comment]
> Looks good to me, I'm assuming you tested them, right?

Yes, only on one system though...

File: trunk/problems/MAXWELL/GALERKIN/makefile (hunk 1 above)
Comment (stefanhenneking) [reply trunk/problems/MAXWELL/GALERKIN/makefile:16]: We can add that, yes. I think we should probably move all of these additional FFLAGS into m_options files and then read them from there only.

**Pull request merged**